src/elements/common/content-answers/ContentAnswersModal.tsx

```diff
@@ -0,0 +1,199 @@
+            const updatedLastQuestion = { ...lastQuestion, ...question };
+            return [...prevState.slice(0, -1), updatedLastQuestion];
+        });
+    }, []);
+
+    const handleErrorCallback = useCallback((error: ElementsXhrError, question: QuestionType): void => {
+        const rateLimitingRegex = /Too Many Requests/i;
+
+        const errorMessage = error?.message || '';
+        const isRateLimitingError = error?.response?.status === 429 || rateLimitingRegex.test(errorMessage);
+
+        const errorQuestion = {
+            ...question,
+            error: isRateLimitingError ? ANSWER_ERROR.RATE_LIMITING : ANSWER_ERROR.GENERAL,
+            isCompleted: true,
+            isLoading: false,
+        };
+
+        setQuestions(prevState => {
+            return [...prevState.slice(0, -1), errorQuestion];
+        });
+    }, []);
+
+    const handleAsk = useCallback(
+        async (question: QuestionType, aiAgent: AgentType, isRetry = false) => {
+            if (onAsk) {
+                onAsk();
+            }
+            const { id } = file;
+            const items = [
+                {
+                    id,
+                    type: 'file',
+                },
+            ];
+            question.isCompleted = false;
+            question.isLoading = true;
+
+            const dialogueHistory = questions.map(q => ({
+                prompt: q.prompt,
+                answer: q.answer,
+                created_at: q.created_at,
+            }));
+
+            setQuestions(prevQuestions => {
+                return [...(isRetry ? prevQuestions.slice(0, -1) : prevQuestions), question];
+            });
+            setIsLoading(true);
+
+            try {
+                const response = await api
+                    .getIntelligenceAPI(true)
+                    .ask(question, items, dialogueHistory, { include_citations: isCitationsEnabled });
+                handleSuccessCallback(response);
+            } catch (e) {
+                handleErrorCallback(e, question);
+            }
+            setIsLoading(false);
+        },
+        [api, file, handleErrorCallback, handleSuccessCallback, isCitationsEnabled, onAsk, questions],
+    );
+
+    const handleRetry = useCallback(
+        (question: QuestionType) => {
+            setQuestions(prevState => {
+                delete question.error;
+                return [...prevState.slice(0, -1), question];
+            });
+            handleAsk(question, null, true);
+        },
+        [handleAsk],
+    );
+
+    const handleClearConversation = useCallback(() => {
+        if (onClearConversation) {
+            onClearConversation();
+        }
+        setQuestions([]);
+    }, [onClearConversation]);
+
+    const handleOnRequestClose = useCallback(() => {
+        if (onRequestClose) {
+            onRequestClose();
+        }
+    }, [onRequestClose]);
+
+    const fileName = getProp(file, 'name');
+    const currentFileExtension = getProp(file, 'extension');
```

Comment:
nit: `current` seems unneeded
```suggestion
    const fileExtension = getProp(file, 'extension');
```